src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3508,12 +3511,17 @@ CoPartitionedTables(Oid firstRelationId, Oid secondRelationId)
 		return true;
 	}
 
+	typeId = sortedFirstIntervalArray[0]->valueTypeId;
```

Comment:
We don't need to resolve the function anymore. `firstTableCache ` and/or `secondTableCache ` has already a field called `shardIntervalCompareFUnction` which holds the compare function. 
(Btw, for instance the code here does not consider hash partitioned tables.)